pkg/executor/compact_table.go

```diff
@@ -257,33 +257,38 @@ func (task *storeCompactTask) compactOnePhysicalTable(physicalTableID int64) (bo
 		if err != nil {
 			// Even after backoff, the request is still failed.., or the request is cancelled or timed out
 			// For example, the store is down. Let's simply don't compact other partitions.
-			warn := errors.Errorf("compact on store %s failed: %v", task.targetStore.Address, err)
+			warn := errors.NewNoStackErrorf("compact on store %s failed: %v", task.targetStore.Address, err)
 			task.parentExec.Ctx().GetSessionVars().StmtCtx.AppendWarning(warn)
+			newErr := errors.Trace(warn)
```

Comment:
what's the difference

Reply:
AppendWarning(warn) will be stack free

Reply:
from the stack info itself, the only difference is the come-out line number of deepest frame